demos/https/iot_demo_https_s3_download_sync.c

```diff
@@ -60,6 +60,8 @@
  * bytes in a file with the header: "Range: bytes=N-M", where N is the starting range and M is the ending range. The
  * S3 HTTP server will response with a 206 Partial Content type of response and the file byte range requested. Please
  * note that not all HTTP servers support a Partial Content download with a byte range.
+ *
+ * This demo cannot download a file larger than 2^32 - 1 bytes.
```

Comment:
Should this be 2^31 - 1? Because you're using signed integer for range.

Reply:
didn't see you changed to unsigned, nvm.

Reply:
Thanks for review!